src/_pytest/config/__init__.py

```diff
@@ -1182,7 +1182,12 @@ def _validate_plugins(self) -> None:
     def _warn_or_fail_if_strict(self, message: str) -> None:
         if self.known_args_namespace.strict_config:
             fail(message, pytrace=False)
-        sys.stderr.write("WARNING: {}".format(message))
+
+        from _pytest.warnings import _issue_warning_captured
+
+        _issue_warning_captured(
+            PytestConfigWarning(message), self.hook, stacklevel=2,
```

Comment:
I'm always not 100% which value it should be, so just making sure, did you check the `stacklevel=2` is the best value here? I think it would be best to point at the caller of `_warn_or_fail_if_strict`.

Reply:
the caller of `_validate_keys` is `_preparse`, which already had `stacklevel=2`, so we'd want 2 here as well right @bluetech ?